src/libawkward/array/ListOffsetArray.cpp

```diff
@@ -2074,22 +2074,25 @@ namespace awkward {
 
       ContentPtr trimmed = content_.get()->getitem_range_nowrap(globalstart,
                                                                 globalstop);
-
       ContentPtr outcontent = trimmed.get()->argsort_next(
-        negaxis, util::make_starts(offsets_), nextparents,
-        offsets_.length() - 1, ascending, stable, false);
+        negaxis,
+        util::make_starts(offsets_),
+        shifts,
+        nextparents,
+        offsets_.length() - 1,
+        ascending,
+        stable);
+
+      Index64 outoffsets(offsets_.length());
+      // FIXME: move to kernel
+      for (int64_t i = 0; i < offsets_.length(); i++) {
+        outoffsets.data()[i] = offsets_.data()[i] - offsets_.data()[0];
+      }
 
```

Comment:
@agoose77 - same here: a temporary fix - I left it for you to provide a proper one :-)